src/Pim/PimMain.tsx

```diff
@@ -67,8 +67,10 @@ class PimMain extends React.PureComponent<PropsType> {
     // FIXME:Hack
     const itemUid = `${(event as any).journalUid}|${encodeURIComponent(event.uid)}`;
 
-    this.props.history!.push(
-      routeResolver.getRoute('pim.events._id', { itemUid }));
+    if (event.uid) {
```

Comment:
I think it should be clickable, just the "edit" button shouldn't be clickable in the event page itself.

Reply:
How should it handle the journalUid being undefined?

Reply:
That's actually another thing I wanted to raise, but thought maybe I would do it in a future PR, though I guess it's already an issue!
I think the birthday calendar should be added earlier in the chain. You are now adding it as part of the events, but I think it should be added as a fake journal altogether. Adding it there is not that different, you just need to add a new calendar journal when decrypting the data (SyncGate.tsx? not sure), make it read-only, and set the color and name as you want it. That's it. :)